### DISCUSSION TITLE
Windows 11 and 10 Evaluation builds fail to start

### DISCUSSION THREAD

**Discussion by ahb2k2k** [vmware/packer-examples-for-vsphere#884] [Help] [answered]
hello,

I'm very new to packer and devops in general and trying to learn through this tool. I was able to figure out some issues like:
- VM was created but CD rom for iso was not connected.
- Kept getting a bad CD error when I cancelled the process with the 'c' option when using ./build.sh

However, I am now facing an issue where it looks like the autounattend.xml file is not being loaded. I am using the EVAL version on windows 10. I also tried the change someone else mentioned here a few days ago about editing the autounattend.pkrtpl.hcl but that didn't resolve my issue.

Note: I do have a valid win 10 ent key which I had initially entered but that isn't working

I'd appreciate any assistance.

**Comment by tenthirtyam** [comment]
Assume you have the require binary to create the ISO on demand and also that your host has vmtools?

**Comment by ahb2k2k** [comment]
I grabbed the ISO from the evaluation center and uploaded it to a datastore.
I’m not connecting to a specific host but to the cluster and yes vmtools are in the same folder as the ISOs.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by tenthirtyam** [reply]
An ISO should be generated by Packer and uploaded to the designated data store. This ISO, along with the VMware Tools ISO **on the ESXi host**, and the OS ISO should be mounted. Total: 3.

If not, would need to know what OS, from the list tested by the project, is your Packer host.

Assume that the "stuck" is the first screen and does not proceed to any installation step. Any error message?

**Reply by ahb2k2k** [reply]
I’m a bit confused. I followed all the steps in the documentation : https://vmware-samples.github.io/packer-examples-for-vsphere/getting-started/build/?darkschemeovr=1

I do see 3 CD-ROMs are attached to the VM. My packer host is Ubuntu. 

If I understand correctly, packer will generate an ISO for me and convert it to template (or OVF for the content library). However, I’m stuck at the stage where it starts the process to generate the ISO. So I see it mounts stuff, creates a folder on the datastore and I can see the VM has been turned on to start provisioning it and doing everything else. But it gets stuck at the installation screen as I don’t think it can see the autounattend file. 

Would it help if I posted my config file?

**Reply by tenthirtyam** [reply]
If you like. Use a Gist link.

**Reply by ahb2k2k** [reply]
https://gist.github.com/ahb2k2k/8cb71c922ac5ba5605d569bb36bda6f8

Additionally, this line here _`iso_content_library_item = ""`_ in the windows-desktop-10.pkrvars.hcl I had to leave blank or it would do a double path when executing.

![packer_error](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/3e45b038-2da8-48d4-a79a-7b32d5a84d39)

**Reply by tenthirtyam** [reply]
Looks like it simply can't find the ISO on the provided datastore and path.

**Reply by ahb2k2k** [reply]
I got that part resolved after leaving iso_content_library empty. I can tell because it gets me to the install screen for windows 10. Do the config files look ok?

**Reply by tenthirtyam** [reply]
`common_iso_content_library` is only activity used if `common_iso_content_library_enabled` is `true`.

Otherwise, the configuration files look ok.

What's the current error from the build and on the VM console?

**Reply by ahb2k2k** [reply]
There is no error. This is where it gets stuck on the console and the build.

I see 3 CD Drives:

1- [abc-str1-vmfs-prod-test] ISO/Windows_10_Enterprise.iso
2. [] /vmimages/tools-isoimages/windows.iso
3. [abc-str1-vmfs-prod-test] packer_cache/packer3443947087.iso

In my datastore, the files (ISOs) I have are listed.

![Packer_running_stuck](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/efd370be-7a5d-4695-b4e2-de0986d15856)
![console](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/fc34f0a4-a9e3-4870-853d-486d1f2d57c0)
![iso_datastore](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/f5238910-dbc7-4f43-9371-e07aebc371b2)

**Reply by tenthirtyam** [reply]
Do you get the same issue with 2022 or is it desktop specific?

**Reply by ahb2k2k** [reply]
Just confirmed it's working for 2022. At least its getting beyond the install splash screen.

Will report back when it completes. Any ideas why the desktop wouldn't work?

**Reply by ahb2k2k** [reply]
Build failed with another error now. 
![server_build_error](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/38a46e63-3cca-4e0a-b299-f1637d9c93e2)

**Reply by tenthirtyam** [reply]
The winrm error is related to your ansible controller.

```
pip3 install pywinrm
```

**Reply by tenthirtyam** [reply]
Could be your Win10 ISO but the evaluation should have no issue. I'd need to reproduce time permitting.

**Reply by ahb2k2k** [reply]
I would appreciate that, thank you.

I will download the ISO again from the eval center.

**Reply by ahb2k2k** [reply]
@tenthirtyam you are the man! 

installed pywinrm as you indicated and the build was successful for Win server 2022. I will attempt Win 10 and 11 tomorrow after downloading new ISOs.

For win 11, is there anything I need to do for vTPM or is that all included?

**Reply by tenthirtyam** [reply]
Enable a key provider in vSphere. The native will work.

**Reply by ahb2k2k** [reply]
Were you able to test on your end? I just tried it using a fresh download of the ISO from the eval center for windows 10 enterprise and still experiencing the same issue.

**Reply by tenthirtyam** [reply]
Will look at it this week.

**Reply by ahb2k2k** [reply]
Thank you. I suspect it's something very dumb that I messed up. 

This is where it gets stuck. If I manually click on 'Next' from the vmware console, it goes through the entire build flawlessly. Is it not finding the image name for some reason in the autounattend file? 

I have compared the settings to the working version of windows server 2022 autounattend and the lines below are the same.

```
<InstallFrom>
                  <MetaData wcm:action="add">
                     <Key>/IMAGE/NAME</Key>
                     <Value>${vm_inst_os_image}</Value>
                  </MetaData>
</InstallFrom>
```

![win_10_error](https://github.com/vmware-samples/packer-examples-for-vsphere/assets/167465956/1b3e1911-54c5-4249-a0a6-a2dcce7305c2)

**Reply by tenthirtyam** [reply]
Ah, i see the problem! 🤦‍♂️

I'll get a fix posted.

**Reply by ahb2k2k** [reply, marked as answer]
UPDATE: I edited the following line in the  windows.pkr.hcl file and that fixed my problem:
`vm_inst_os_image_ent = "Windows 10 Enterprise"` to the following:
`vm_inst_os_image_ent = "Windows 10 Enterprise Evaluation" `

**Reply by tenthirtyam** [reply]
Yep, looks like I'll need to configure it to amend "Evaluation" if eval is true.

**Reply by tenthirtyam** [reply]
Replace the following:

`vm_inst_os_image = var.vm_inst_os_image_pro`

with:

`vm_inst_os_image = var.vm_inst_os_eval ? "${var.vm_inst_os_image_ent} Evaluation" : var.vm_inst_os_image_pro`

And then...

`vm_inst_os_image = var.vm_inst_os_image_ent`

with:

`vm_inst_os_image = var.vm_inst_os_eval ? "${var.vm_inst_os_image_ent} Evaluation" : var.vm_inst_os_image_ent`

**Reply by tenthirtyam** [reply]
#896 opened.
